src/test-rng.cpp

```diff
@@ -0,0 +1,55 @@
+/*
+ * This file uses the Catch unit testing library, alongside
+ * testthat's simple bindings, to test a C++ function.
+ *
+ * For your own packages, ensure that your test files are
+ * placed within the `src/` folder, and that you include
+ * `LinkingTo: testthat` within your DESCRIPTION file.
+ */
+
+// All test files should include the <testthat.h>
+// header file.
+#include <testthat.h>
+#include "rng.h"
+#include <Rcpp.h>
+
+void set_seed(double seed) {
+    Rcpp::Environment base_env("package:base");
+    Rcpp::Function set_seed_r = base_env["set.seed"];
+    set_seed_r(std::floor(std::fabs(seed)));
+}
+
+bool almost_equal(double a, double b, double delta) {
+	return a < b + delta && a > b - delta;
+}
+
+context("Random number generation") {
```

Comment:
This is the first bit of C++/R testing I've seen and it looks nice! Would you mind writing a blog post about it perhaps?

Reply:
Sure!